irohad/multi_sig_transactions/transport/impl/mst_transport_grpc.cpp

```diff
@@ -75,9 +77,24 @@ grpc::Status MstTransportGrpc::SendState(
 
   for (auto &batch : batches) {
     batch_factory_->createTransactionBatch(batch).match(
-        [&](iroha::expected::Value<
-            std::unique_ptr<shared_model::interface::TransactionBatch>>
-                &value) { new_state += std::move(value).value; },
+        [&](iroha::expected::Value<std::unique_ptr<
+                shared_model::interface::TransactionBatch>> &value) {
+          auto cache_presence = tx_presence_cache_->check(*(value.value));
+          auto is_replay = std::any_of(
```

Comment:
Though it's really not scope of this PR, I would like to mention, that this method should be in interface as well as the `check` one. In future, I think, the same algorithm could be used more than once, and this will lead to code duplication, which is not good, taking into account that it can be easily transferred to a separate function